src/ControlSystem/Systems/Shape.hpp

```diff
@@ -69,8 +69,9 @@ struct Shape : tt::ConformsTo<protocols::ControlSystem> {
     const auto compact_index = iter.compact_index(i);
     if (compact_index.has_value()) {
       iter.set(compact_index.value());
-      return {"(l,m)=("s + get_output(iter.l()) + ","s + get_output(iter.m()) +
-              ")"s};
+      const int m = i < num_components / 2 ? static_cast<int>(iter.m())
+                                           : -static_cast<int>(iter.m());
+      return {"(l,m)=("s + get_output(iter.l()) + ","s + get_output(m) + ")"s};
```

Comment:
Maybe delete the "(l,m)=" and the parens, and just return "l,m"? Then I think you end up with names like `Shape_2,2` in the output which is intuitive enough I think.

Reply:
With my current changes, the h5 file will look like:
```
/ControlSystem/Shape/(l,m)=(0,0).dat
/ControlSystem/Shape/(l,m)=(1,0).dat
/ControlSystem/Shape/(l,m)=(1,1).dat
/ControlSystem/Shape/(l,m)=(1,-1).dat
```

With your suggestion, it'll look like
```
/ControlSystem/Shape/0,0.dat
/ControlSystem/Shape/1,0.dat
/ControlSystem/Shape/1,1.dat
/ControlSystem/Shape/1,-1.dat
```

I think the second method looks a bit strange, but I don't have a strong preference

Reply:
How about `Shape/l2m2.dat`?

Reply:
I like it! I'll do that